docs(schema): add descriptions for configuration properties in schema

skrashevich opened the pull request.

AlexxIT commented:
There wasn't any particular logic there. It's just that now it's very hard to compare before and after.

skrashevich commented:
> There wasn't any particular logic there. It's just that now it's very hard to compare before and after.

  ### Definitions Section                                                     
                                                                              
  1. Field listen:                                                            
    • Added description: "Specifies the address and port on which the server  
    will listen for incoming connections."                                    
  2. Field log_level:                                                         
    • Added description: "Defines the logging level for different components."
                                                                              
                                                                              
  ### Properties Section                                                      
                                                                              
  #### API Configuration                                                      
                                                                              
  1. Added description: "Configuration for the API server."                   
  2. base_path:                                                               
    • Added description: "Base path for the API endpoints."                   
  3. listen:                                                                  
    • Added description: "Specifies the address and port on which the API     
    server will listen."                                                      
  4. origin:                                                                  
    • Added description: "Allowed origin for CORS. Use '*' to allow all       
    origins."                                                                 
  5. password:                                                                
    • Added description: "Password for API authentication."                   
  6. static_dir:                                                              
    • Added description: "Directory for serving static files."                
  7. tls_cert:                                                                
    • Added description: "TLS certificate file."                              
  8. tls_key:                                                                 
    • Added description: "TLS private key file."                              
  9. tls_listen:                                                              
    • Added description: "Specifies the address and port for TLS connections."
  10. unix_listen:                                                            
    • Added description: "Specifies the Unix socket on which the server will  
    listen."                                                                  
  11. username:                                                               
    • Added description: "Username for API authentication."                   
                                                                              
                                                                              
  #### FFmpeg Configuration                                                   
                                                                              
  1. Added description: "Configuration for FFmpeg integration."               
  2. bin:                                                                     
    • Added description: "Path to the FFmpeg binary."                         
                                                                              
                                                                              
  #### Home Assistant Configuration (hass)                                    
                                                                              
  1. Added description: "Configuration for Home Assistant integration."       
  2. config:                                                                  
    • Added description: "Path to the Home Assistant configuration directory."
                                                                              
                                                                              
  #### HomeKit Configuration                                                  
                                                                              
  1. Added description: "Configuration for HomeKit integration."              
  2. device_id:                                                               
    • Added description: "Device ID for the HomeKit device."                  
  3. device_private:                                                          
    • Added description: "Private key for the HomeKit device."                
  4. name:                                                                    
    • Added description: "Name of the HomeKit device."                        
  5. pairings:                                                                
    • Added description: "List of paired devices."                            
  6. pin:                                                                     
    • Added description: "PIN for pairing with HomeKit devices."              
                                                                              
                                                                              
  #### Logging Configuration                                                  
                                                                              
  1. Added description: "Configuration for logging."                          
  2. exec:                                                                    
    • Added description: "Log level for exec module. Value 'debug' will print 
    stderr."                                                                  
  3. ffmpeg:                                                                  
    • Added description: "Log level for FFmpeg. Will only be displayed with   
    'exec: debug' setting."                                                   
  4. format:                                                                  
    • Added description: "Format of the logs. Can be 'color', 'json', or      
    'text'."                                                                  
  5. level:                                                                   
    • Added description: "Default log level for all components."              
  6. output:                                                                  
    • Added description: "Output destination for the logs. Can be 'stdout',   
    'stderr', or a file path."                                                
  7. time:                                                                    
    • Added description: "Timestamp format for the logs."                     
                                                                              
                                                                              
  #### Ngrok Configuration                                                    
                                                                              
  1. Added description: "Configuration for ngrok integration for external     
  access."                                                                    
  2. command:                                                                 
    • Added description: "Command to run ngrok."                              
                                                                              
                                                                              
  #### Publishing Configuration                                               
                                                                              
  1. Added description: "Configuration for publishing streams to external     
  services."                                                                  
  2. additionalProperties:                                                    
    • Added description: "URL for publishing the stream."                     
                                                                              
                                                                              
  #### RTMP Configuration                                                     
                                                                              
  1. Added description: "Configuration for RTMP server."                      
  2. listen:                                                                  
    • Added description: "Specifies the address and port for RTMP             
    connections."                                                             
                                                                              
                                                                              
  #### RTSP Configuration                                                     
                                                                              
  1. Added description: "Configuration for RTSP server."                      
  2. default_query:                                                           
    • Added description: "Default query parameters for RTSP streams."         
  3. listen:                                                                  
    • Added description: "Specifies the address and port for RTSP             
    connections."                                                             
  4. password:                                                                
    • Added description: "Password for RTSP authentication."                  
  5. pkt_size:                                                                
    • Added description: "Packet size for RTSP streams."                      
  6. username:                                                                
    • Added description: "Username for RTSP authentication."                  
                                                                              
                                                                              
  #### SRTP Configuration                                                     
                                                                              
  1. Added description: "Configuration for SRTP server for HomeKit."          
  2. listen:                                                                  
    • Added description: "Specifies the address and port for SRTP             
    connections."                                                             
                                                                              
                                                                              
  #### Streams Configuration                                                  
                                                                              
  1. Added description: "Configuration for stream sources."                   
  2. additionalProperties:                                                    
    • Added description: "Source URL or configuration for the stream."        
                                                                              
                                                                              
  #### WebRTC Configuration                                                   
                                                                              
  1. Added description: "Configuration for WebRTC server."                    
  2. candidates:                                                              
    • Added description: "List of additional host candidates for WebRTC       
    connections."                                                             
  3. filters:                                                                 
    • Added description: "Filters for auto discovery logic in WebRTC."        
  4. ice_servers:                                                             
    • Added description: "List of STUN and TURN servers for WebRTC."          
  5. listen:                                                                  
    • Added description: "Specifies the address and port for WebRTC           
    connections."                                                             
                                                                              
                                                                              
  #### WebTorrent Configuration                                               
                                                                              
  1. Added description: "Configuration for WebTorrent integration."           
  2. shares:                                                                  
    • Added description: "Configuration for sharing torrents."                
  3. trackers:                                                                
    • Added description: "List of WebTorrent trackers."                       
                                                                              
                                                                              
  ### Overall File Metadata                                                   
                                                                              
  1.  $schema  remains unchanged.                                             
  2. Title changed to "go2rtc"                                                
  3. Type remains "object"

skrashevich commented:
#1980
